docs/windows.md

```diff
@@ -1,142 +1,141 @@
+[Cygwin→](#cygwin)  
 
-Use the following command to download go-ipfs source code:
+**Using build tools manually:**  
+This section assumes you have a working version of `go` and `git` already setup. You may want to build this way if your environment restricts installing additional software, or if you're integrating IPFS into your own build system.  
+[Minimal→](#minimal)  
 
+## MSYS2
+1. Install msys2 (http://www.msys2.org)  
+2. Run the following inside a normal `cmd` prompt (Not the MSYS2 prompt, we only need MSYS2's tools).  
+An explanation of this block is below.
 ```
+SET PATH=%PATH%;\msys64\usr\bin
+pacman --noconfirm -S  git make unzip
 go get -u github.com/ipfs/go-ipfs
+cd %GOPATH%\src\github.com\ipfs\go-ipfs
+make install
+%GOPATH%\bin\ipfs.exe version --all
 ```
 
-The above command uses Git to download go-ipfs from its GitHub
-repository. If you get authentication problems with Git, you might
-want to take a look at
-https://help.github.com/articles/caching-your-github-password-in-git/
-and use the suggested solution:
-
-```
-git config --global credential.helper wincred
-```
+If there were no errors, the final command should output version information similar to "`ipfs version 0.4.14-dev-XXXXXXX`" where "XXXXXXX" should match the current short-hash of the `go-ipfs` repo. You can retrieve said hash via this command: `git rev-parse --short HEAD`. If `ipfs` runs and the version string matches, building was successful.
 
-## Choose the way you want to proceed
-
-Now there are two ways to download, install the dependencies and to
-build go-ipfs:
-
-1) There is the "Manual Way", where you don't need to install anymore
-software except the dependencies, but you have a number of commands to
-type.
-
-2) There is a way by installing 'make' through Cygwin and using it to
-do nearly everything. We call this way the "Cygwin Way". It may take
-much more time, because installing Cygwin can take a lot of time, but
-after that it might be easier as many procedures are just a 'make'
-command away.
-
-So use the next steps below that start with "Manual Way" if that's the
-way you want, otherwise scroll down a bit and use the "Cygwin Way"
-steps below.
-
-## Manual Way: download and install dependencies
-
-The following commands should download or update go-ipfs dependencies
-and then install them:
+|Command|Explanation|
+| ---: | :--- |
+|`SET PATH=%PATH%;\msys64\usr\bin`         |Add msys2's bin to `PATH`; Defaults to: (\msys64\usr\bin)|
+|`pacman --noconfirm -S  git make unzip`   |Install `go-ipfs` build dependencies|
+|`go get -u github.com/ipfs/go-ipfs`       |Fetch / Update `go-ipfs` source|
+|`cd %GOPATH%\src\github.com\ipfs\go-ipfs` |Change to `go-ipfs` directory|
+|`make install`                            |Build and install to `%GOPATH%\bin\ipfs.exe`|
+|`%GOPATH%\bin\ipfs.exe version --all`     |Test the built binary|
 
+**Tip**: You can permanently append the MSYS2 bin to your `PATH` so that on subsequent builds you can simply run
```

Comment:
Because of the way `SETX` works, I've changed the 'tip' bit a little.
`SETX` doesn't actually set the variable for the current session, so what we want to do is, first `SET` `PATH` and then lock that in with `SETX`. In addition we don't need to specify the msys|cygwin path since it's already contained in `%PATH%`, the way it was written before could lead the user to accidentally append it twice if it was already set, now we expect it to be set ahead of time and optionally locked that way.
***
**Tip**: To avoid setting `PATH` every time (`SET PATH=%PATH%;\msys64\usr\bin`), you can lock it in permanently using `setx` after it's been set once:
```
SETX PATH %PATH%
```